scripts/build.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Builds all features of the specified package using Cargo.
+# Usage:
+#   build.sh
+#
+# This script extracts the features defined in the package's Cargo.toml file and builds each one individually using Cargo.
+# It is designed for use with crates that may have multiple conditional compilation features, like `containerd-shim-wasm`.
+#
+# The script takes the following actions for each feature:
+#  * Echoes the feature name being built.
+#  * Runs the cargo build command specifically for that feature, with no default features.
+#
+# The script requires the following variables to be set in the script itself:
+#  * FEATURES_FILE: The path to the Cargo.toml file containing the features definitions. (e.g. "crates/containerd-shim-wasm/Cargo.toml")
+#  * PACKAGE: The name of the Rust package to build. (e.g. "containerd-shim-wasm")
+#
+# This script is intended to be run within the project directory, where the Cargo.toml file is located.
+
+set -euo pipefail
+
+FEATURES_FILE="crates/containerd-shim-wasm/Cargo.toml"
+PACKAGE="containerd-shim-wasm"
+
+# Check if FEATURES_FILE exists
+if [[ ! -f "$FEATURES_FILE" ]]; then
+  echo "File $FEATURES_FILE not found."
+  exit 1
+fi
+
+# captures [features] section, filters out [features], splits on =, trims whitespace
+FEATURES=$(sed -n '/\[features\]/,/^\[.*\]/p' "$FEATURES_FILE" | grep -v '^\[' | cut -d '=' -f 1 | tr -d ' ')
```

Comment:
Have you see `cargo-whatfeatures`?

```console
$ cargo whatfeatures -t --json ./crates/cargo whatfeatures -t --json ./crates/containerd-shim-wasmedge/ | jq .members[0].features
{
  "default": [
    "standalone",
    "static"
  ],
  "standalone": [
    "wasmedge-sdk/standalone"
  ],
  "static": [
    "wasmedge-sdk/static"
  ],
  "wasi_nn": [
    "wasmedge-sdk/wasi_nn"
  ]
}
```

There is also https://github.com/frewsxcv/cargo-all-features specifically for building/testing all feature combos.

Reply:
cargo-all-features looks pretty awesome